src/MultiMessageReceiver.sol

```diff
@@ -138,6 +153,9 @@ contract MultiMessageReceiver is IMultiMessageReceiver, ExecutorAware, Initializ
         /// increment quorum
         ++messageVotes[msgId];
 
+        /// stores the delivery time
+        deliveryTime[msgId] = block.timestamp;
```

Comment:
Minor:

Here the `deliveryTime` is updated every time when the message is received from a single adapter. Ideally, it should only be updated when the quorum is reached. 

For example, let's say the quorum is 2/3, and the delay period is 3 days, two bridges delivered the message immediately, then the message should be able to be executed 3 days later. Now if the third bridge delivered the message 3 days later right before the message got executed, according to the current code, the message has to wait for another 3 days to get executed, while IMO it does not need to.